pkg/flist/flist.go

```diff
@@ -438,18 +438,29 @@ func (f *flistModule) downloadFlist(url string) (string, error) {
 	hash, err := f.FlistHash(url)
 	if err == nil {
 		flistPath := filepath.Join(f.flist, strings.TrimSpace(string(hash)))
-		_, err = os.Stat(flistPath)
+		f, err := os.Open(flistPath)
 		if err != nil && !os.IsNotExist(err) {
 			return "", err
 		}
+		defer f.Close()
```

Comment:
I'm not sure about this. Cause it could be that the `f.saveFlist(resp.Body)` is called while the file is still open, and there is a rename call into this function. I'm not sure if this is a problem or not.